src/App.jsx

```diff
@@ -7,19 +7,35 @@ import database from './database.json';
 import './App.css';
 
 function App() {
+  const [productsList, setProductsList] = useState(database);
+
+  const upvoteProduct = (id) => {
+    const productListClone = [...productsList];
+    const elementIndex = productListClone.findIndex((el) => el.id === id);
+
+    productListClone[elementIndex] = {
+      ...productListClone[elementIndex],
+      upvotes: productListClone[elementIndex].upvotes + 1,
+    };
```

Comment:
Replace the object in the list by a exact copy of it but adding a single unit to the `upvotes` counter.